vignettes/sliding.Rmd

```diff
@@ -0,0 +1,281 @@
+
+```{r grab-epi-data}
+theme_set(theme_bw())
+#%% Remove below covidcast code and library(covidcast) 
+# y <- covidcast_signals(
+#   c("doctor-visits", "jhu-csse"),
+#   c("smoothed_adj_cli", "confirmed_7dav_incidence_prop"),
+#   start_day = "2020-06-01", 
+#   end_day = "2021-12-01",
+#   issues = c("2020-06-01", "2021-12-01"),
+#   geo_type = "state", 
+#   geo_values = c("ca", "fl")) 
+# saveRDS(y, "/inst/extdata/ca_fl_covidcast_signals.rds")
+y <- readRDS(
+  system.file("extdata", "ca_fl_covidcast_signals.rds", package = "epipredict", mustWork = TRUE)
+)
+ 
+y1 <- y[[1]] %>% 
+  select(geo_value, time_value, version = issue, percent_cli = value) %>%
+  as_epi_archive()
+
+x <- epix_merge(
+  y1, y[[2]] %>% 
+    select(geo_value, time_value, version = issue, case_rate = value) %>%
+    as_epi_archive(), sync = "locf")
+```
+
+After obtaining the latest snapshot of the data, we produce forecasts on that data using the default engine of simple linear regression and compare to a random forest (as were introduced in the Getting started with epipredict vignette). #%% Add link to getting started vignette once html file is there!
+
+Note that all of the warnings about `forecast_date` being less than the most recent update date of the data have been suppressed to avoid cluttering the output.
+
+```{r make-arx-kweek, warning = FALSE}
+# Latest snapshot of data, and forecast dates
+x_latest <- epix_as_of(x, max_version = max(x$versions_end)) 
+fc_time_values <- seq(as.Date("2020-08-01"), as.Date("2021-12-01"), 
+                      by = "1 month")
+
+k_week_ahead <- function(ahead = 7, engine_type = "linear_reg") {
+  if(engine_type == "linear_reg"){
+    x_latest %>%
+      epi_slide(function(x, ...) 
+        arx_forecaster(x, outcome = "case_rate", 
+                       predictors = c("case_rate", "percent_cli"),  
+                       args_list = arx_args_list(ahead = ahead))$predictions %>% 
+          select(-c(geo_value, time_value)),
+        n = 120, ref_time_values = fc_time_values, new_col_name = "fc") %>% 
+      mutate(engine_type = "linear_reg")
+  } else if (engine_type == "random_forest"){
+    x_latest %>%
+      epi_slide(function(x, ...) 
+        arx_forecaster(x, outcome = "case_rate", 
+                       predictors = c("case_rate", "percent_cli"),
+                       rand_forest(mode = "regression"), 
+                       args_list = arx_args_list(ahead = ahead))$predictions %>% 
+          select(-c(geo_value, time_value)),
+        n = 120, ref_time_values = fc_time_values, new_col_name = "fc") %>% 
+      mutate(engine_type = "random_forest")
+  } 
+}
+
+# Generate the forecasts, and bind them together
+fc <- bind_rows(
+  purrr::map_dfr(c(7,14,21,28), ~ k_week_ahead(.x, engine_type = "linear_reg")),
```

Comment:
`map_dfr` is in the `epipredict` NAMESPACE, so I don't think you need the `purrr` version.

Reply:
I don't see that fun. in the `epipredict` NAMESPACE yet?

Reply:
You're right. It's internal and not exported. NM.